character-controller.js

```diff
@@ -991,13 +993,18 @@ class UninterpolatedPlayer extends StatePlayer {
       crouch: new BiActionInterpolant(() => this.hasAction('crouch'), 0, crouchMaxTime),
       activate: new UniActionInterpolant(() => this.hasAction('activate'), 0, activateMaxTime),
       use: new InfiniteActionInterpolant(() => this.hasAction('use'), 0),
+      useEnvelope: new BiActionInterpolant(() => this.getAction('use')?.animationEnvelope?.length > 0, 0, 200),
       pickUp: new InfiniteActionInterpolant(() => this.hasAction('pickUp'), 0),
       unuse: new InfiniteActionInterpolant(() => !this.hasAction('use'), 0),
       aim: new InfiniteActionInterpolant(() => this.hasAction('aim'), 0),
       aimRightTransition: new BiActionInterpolant(() => this.hasAction('aim') && this.hands[0].enabled, 0, aimTransitionMaxTime),
       aimLeftTransition: new BiActionInterpolant(() => this.hasAction('aim') && this.hands[1].enabled, 0, aimTransitionMaxTime),
       narutoRun: new InfiniteActionInterpolant(() => this.hasAction('narutoRun'), 0),
       fly: new InfiniteActionInterpolant(() => this.hasAction('fly'), 0),
+      flyDash: new BiActionInterpolant(() => {
+        const ioManager = metaversefile.useIoManager();
+        return this.hasAction('fly') && ioManager.keys.shift;

```

Comment:
In general, can we avoid testing `ioManager` in character controller?

I think for supporting NPCs we can't really check the `ioManager`, since that is tied to the local user input, and it will interfere with other characters.

Reply:
Calc `mirrorFactor` from `avatar.getAngle()` instead of `ioManager`. https://github.com/upstreet-labs/app/pull/82/commits/74a40028a891eaabed973c32bb02b52c5498bd2a

Reply:
Update, now calculating `mirrorFactor` using original methods, from `backwardAnimationSpec`, instead of from `avatar.getAngle()` directly. https://github.com/upstreet-labs/app/pull/82/commits/58797c96e2ede64d7eafdd62e2d0ca21592829a0